agent/api/task/statusmapping.go

```diff
@@ -42,9 +42,9 @@ func MapToTaskStatus(knownState apicontainer.ContainerStatus, steadyState apicon
 	return TaskStatusNone
 }
 
-// ContainerStatus maps the task status to the corresponding container status
-func (ts *TaskStatus) ContainerStatus(steadyState apicontainer.ContainerStatus) apicontainer.ContainerStatus {
-	switch *ts {
+// MapToContainerStatus maps the task status to the corresponding container status
```

Comment:
Why is this not called on `TaskStatus` anymore?

Reply:
ContainerStatus had TaskStatus function, and TaskStatus had ContainerStatus function, making them impossible to separate out into 2 packages, because it's immediately a cyclic import. So I decoupled these functions from ContainerStatus and TaskStatus receivers, and then updated the name to be more explicit, that we are mapping *Status param to the other *Status.

Reply:
sounds good. how about naming it `MapTaskToContainerStatus` to make it more descriptive(since we also have resource status now)?